packages/completer/src/model.ts

```diff
@@ -355,6 +385,33 @@ export class CompleterModel implements Completer.IModel {
     this._stateChanged.emit(undefined);
   }
 
+  /**
+   * Check if CompletionItem matches against query.
+   * Highlight matching prefix by adding <mark> tags.
+   */
+  private _markup(query: string): CompletionHandler.ICompletionItems {
+    let items = JSONExt.deepCopy(this._completionItems.items);
+    let results: CompletionHandler.ICompletionItem[] = [];
+    for (let item of items) {
+      // See if insert text matches query string
+      let match = StringExt.matchSumOfSquares(item.label, query);
+      if (match) {
+        // Highlight label text if there's a match
+        let marked = StringExt.highlight(
+          item.label,
+          match.indices,
+          Private.mark
+        );
+        if (!item.insertText) {
+          item.insertText = item.label;
+        }
```

Comment:
Why do we need this here? I thought in other places it does an || between insertText and label, to choose label if insertText is empty.

Reply:
Ah I see, makes sense, thank you! Could you add a comment here to say that?